rpcs3/rpcs3qt/main_window.cpp

```diff
@@ -85,7 +85,10 @@ void main_window::Init()
 
 	// for highdpi resize toolbar icons and height dynamically
 	// choose factors to mimic Gui-Design in main_window.ui
-	const int toolBarHeight = menuBar()->sizeHint().height() * 1.5;
+	int toolBarHeight = menuBar()->sizeHint().height() * 1.5;
+	if (toolBarHeight == 0) {
```

Comment:
Shouldn't the proper solution be to use `http://doc.qt.io/qt-5.9/qsize.html#isValid`?
I.e. something like 
```cpp
if (menuBar()->sizeHint().isValid()) {
	toolBarHeight = menuBar()->sizeHint().height() * 1.5;
else {
	toolBarHeight = menuBar()->height() * 1.5;
}

Reply:
that would leave us still with no guard for == 0
"Returns true if both the width and height is equal to or greater than 0; otherwise returns false."

Reply:
@Megamouse ah, yeah. Would handle the negative case though. But still, to me it looks like that the solution in this PR is not a proper solution but a hack to workaround some issue somewhere else.

Reply:
Yeah, it's a hack to workaround Ubuntu being awful and weird (or maybe Qt trying to re-act to ubuntu's weirdness).

Reply:
someone with ubuntu:

updateGeometry();

should recalculate the sizehint.
so maybe just call that before.

Reply:
@Megamouse I think it should already be called: https://code.woboq.org/qt5/qtbase/src/widgets/widgets/qmenubar.cpp.html#1611

Reply:
Confirming that adding `updateGeometry();` does not fix the issue on Ubuntu 14.04